README.md

```diff
@@ -76,4 +55,64 @@ in a local experiment:
 |-------------|------------------|------------------|-----------------------------------|
 |  Fastest    |  0.083s          |  0.560s          | 0.556s                            |
 |  Slowest    |  1.078s          |  1.887s          | 2.579s                            |
-|  Average    |  0.580s ± 0.175s |  0.849s ± 0.361s | 1.408s ± 0.542s                   |
\ No newline at end of file
+|  Average    |  0.580s ± 0.175s |  0.849s ± 0.361s | 1.408s ± 0.542s                   |
+
+### Open-Cluster-Management Integration
+
+Cluster-gateway has native integration with [Open-Cluster-Management(OCM)](https://open-cluster-management.io/)
+to provide the KubeVela admin a more coherent user experience in distributing
+applications across multiple clusters:
+
+#### Enabling OCM seamlessly in your KubeVela environment
+
+The official vela addon named [ocm-cluster-manager](https://github.com/oam-dev/catalog/tree/master/addons/ocm-cluster-manager)
+will help you easily bootstrap the OCM control plane (in the hosting cluster
+where your KubeVela control plane lives). Note that the OCM environment 
+installed from the addon above will not take any effect until we opt-in to
+enable the functional integration between KubeVela and OCM as is elaborated 
+below. It's just a minimal trial setup to try out OCM instantly, while in order
+to enable the further integration with OCM, we will need to adjust the 
+configuration of cluster-gateway to make it detect and aware of the local OCM
+environment.
+
+#### Aligning ClusterGateway to OCM's cluster registry
+
+By opt-in to the flag `--ocm-integration=true`, the cluster-gateway will be
+detecting and loading the OCM environment in the hosting cluster and connecting
+each `ClusterGateway` custom resource from cluster-gateway to OCM's original
+cluster model in OCM named `ManagedCluster`. The `ClusterGateway` is a
+gateway "ingress" abstraction for the Kubernetes clusters managed by KubeVela,
+so after integrating with OCM it's intuitive to regard the gateway resource
+as a "satellite" child resource around `ManagedCluster`. Setting the flag will
+make the cluster-gateway filter out those dangling `ClusterGateway` that doesn't
+have a valid `ManagedCluster` bound with. In addition to that, we won't need to
+explicitly set the master URL in the cluster secret because the cluster-gateway
+will be merging the URL list from the corresponding `ManagedCluster`. 
+
+Furthermore, by enabling the integration, we will also reflect/aggregate the 
+healthiness of the corresponding clusters by partially merging the original
```

Comment:
actually it's not yet implemented but we will be getting there real soon. guess it's fine to drop the document before the code.